sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpClientProvider.java

```diff
@@ -13,9 +14,28 @@
  * introduced.
  */
 public final class JdkHttpClientProvider implements HttpClientProvider {
+    // Enum Singleton Pattern
+    private enum GlobalJdkAsyncHttpClient {
+        HTTP_CLIENT(new JdkAsyncHttpClientBuilder().build());
+
+        private HttpClient httpClient;
+
+        GlobalJdkAsyncHttpClient(HttpClient httpClient) {
+            this.httpClient = httpClient;
+        }
+
+        private HttpClient getHttpClient() {
+            return httpClient;
+        }
+    }
 
     @Override
     public HttpClient createInstance() {
-        return new JdkAsyncHttpClientBuilder().build();
+        final String disableDefaultSharingHttpClient =
+            Configuration.getGlobalConfiguration().get("AZURE_DISABLE_DEFAULT_SHARING_HTTP_CLIENT");
```

Comment:
@alzimmermsft @kasobol-msft @JonathanGiles @srnagar  
Do you have a better choice of naming for the `AZURE_DISABLE_DEFAULT_SHARING_HTTP_CLIENT`?

Reply:
I'll actually defer to @lmolkova on this one given her on-going work with configurations in the SDKs to see what best aligns with her future vision

Reply:
Discussed with @lmolkova  offline, `AZURE_DISABLE_HTTP_CLIENT_SHARING` could be a better name.